### PR TITLE
Jetpack Sync: Add sync event for theme deletion

### PR DESCRIPTION
**Pull request by gititon**
This PR adds a Jetpack sync event for theme deletion, primarily for use by the Activity log.

#### Changes proposed in this Pull Request:

Addition of  'jetpack_deleted_theme' action, which triggers sync action upon theme deletion.

#### Testing instructions:

phpunit tests new functionality

Addition of  'jetpack_deleted_theme' action

### CODE REVIEW

File: sync/class.jetpack-sync-module-themes.php [hunk 1]
```diff
@@ -21,6 +23,32 @@ public function init_listeners( $callable ) {
...
+			return;
+		}
+
+		$slug = $delete_theme_call['args'][0];
+
+		/**
+		 * Signals to the sync listener that a theme was deleted and a sync action
+		 * reflecting the deletion and theme slug should be sent
+		 *
+		 * @since 4.9.0
```
Comment (enejb) [inline sync/class.jetpack-sync-module-themes.php]: 4.9 has shipped should be updated to 5.0

**Comment by enejb** [comment]
tests don't quite pass

File: sync/class.jetpack-sync-module-themes.php [hunk 2]
```diff
@@ -11,6 +11,8 @@ public function init_listeners( $callable ) {
 		add_action( 'upgrader_process_complete', array( $this, 'check_upgrader'), 10, 2 );
 		add_action( 'jetpack_installed_theme', $callable, 10, 2 );
 		add_action( 'jetpack_updated_theme', $callable, 10, 2 );
+		add_action( 'delete_site_transient_update_themes', array($this, 'detect_theme_deletion') );
```
Comment (lezama) [inline sync/class.jetpack-sync-module-themes.php]: missing spaces within parenthesis

Comment (lezama) [inline sync/class.jetpack-sync-module-themes.php]: I haven't looked at how the internals of deleting themes work, but why it is only hooked to `delete_site_transient` couldn't happen on `update_site_transient`?

**Comment by dereksmart** [comment]
👍

**Pull request merged**